### ISSUE TITLE
Dav Auto-discover redirects to wrong principal when using LDAP.

### ISSUE THREAD

**Issue by Whoops** [owncloud/core#23498]
### Steps to reproduce
1. Attach Owncloud instance to LDAP
2. Attempt to sync any LDAP user's calendar/contacts via CalDav/CardDav via autodiscovery
### Expected behaviour

Autodiscover should direct to the right Cal/CardDav principal, in my case `a8631c10-474b-1035-947c-83c81fea564e`
### Actual behaviour

Autodiscover redirects based on the login used (in my case `whoops`) instead of principal. I was able to get Calendar working in Thunderbird by manually specifying the URL using the uuid above. I have yet to get it working on Android.. Log from CalDav Sync on Android:

```
20160322T221444Z I org.dmfs.caldav.autheticator.AuthenticatorActivity calendar discovery log:
20160322T221445Z V org.dmfs.dav.Dav get principal /.well-known/caldav
20160322T221446Z V org.dmfs.dav.Dav following redirect to http://cloud.waltonhoops.com/remote.php/dav/
20160322T221447Z V org.dmfs.dav.Dav following redirect to https://cloud.waltonhoops.com/remote.php/dav/
20160322T221448Z I org.dmfs.dav.Dav found principal: /remote.php/dav/principals/users/whoops/
20160322T221448Z V org.dmfs.dav.CalDav get calendars https://cloud.waltonhoops.com/remote.php/dav/principals/users/whoops/
20160322T221448Z V org.dmfs.dav.methods.DavPropFind Status: 404
20160322T221448Z V org.dmfs.dav.methods.DavPropFind <?xml version="1.0" encoding="utf-8"?>
20160322T221448Z V org.dmfs.dav.methods.DavPropFind <d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
20160322T221448Z V org.dmfs.dav.methods.DavPropFind   <s:exception>Sabre\DAV\Exception\NotFound</s:exception>
20160322T221448Z V org.dmfs.dav.methods.DavPropFind   <s:message>Principal with name whoops not found</s:message>
20160322T221448Z V org.dmfs.dav.methods.DavPropFind </d:error>
20160322T221448Z V org.dmfs.dav.CalDav looking for calendars in
20160322T221449Z V org.dmfs.dav.methods.DavPropFind Status: 405
20160322T221449Z V org.dmfs.dav.methods.DavPropFind <html>
20160322T221449Z V org.dmfs.dav.methods.DavPropFind <head><title>405 Not Allowed</title></head>
20160322T221449Z V org.dmfs.dav.methods.DavPropFind <body bgcolor="white">
20160322T221449Z V org.dmfs.dav.methods.DavPropFind <center><h1>405 Not Allowed</h1></center>
20160322T221449Z V org.dmfs.dav.methods.DavPropFind <hr><center>nginx/1.8.1</center>
20160322T221449Z V org.dmfs.dav.methods.DavPropFind </body>
20160322T221449Z V org.dmfs.dav.methods.DavPropFind </html>
20160322T221449Z V org.dmfs.dav.CalDav additionally looking for calendar in
20160322T221449Z V org.dmfs.dav.methods.DavPropFind Status: 405
20160322T221449Z V org.dmfs.dav.methods.DavPropFind <html>
20160322T221449Z V org.dmfs.dav.methods.DavPropFind <head><title>405 Not Allowed</title></head>
20160322T221449Z V org.dmfs.dav.methods.DavPropFind <body bgcolor="white">
20160322T221449Z V org.dmfs.dav.methods.DavPropFind <center><h1>405 Not Allowed</h1></center>
20160322T221449Z V org.dmfs.dav.methods.DavPropFind <hr><center>nginx/1.8.1</center>
20160322T221449Z V org.dmfs.dav.methods.DavPropFind </body>
20160322T221449Z V org.dmfs.dav.methods.DavPropFind </html>

App: org.dmfs.caldav.lib
App version: 0.4.27
OS version: 23
OS Info: google/hammerhead/hammerhead:6.0.1/MMB29V/2554798:user/release-keys
Manufacturer: LGE
Date: Tue Mar 22 16:14:51 MDT 2016
```
### Server configuration

**Operating system**:
FreeBSD 10.2

**Web server:**
Nginx

**Database:**
PostgreSQL

**PHP version:**
7.0

**ownCloud version:** (see ownCloud admin page)
ownCloud 9.0.0 (stable)

**Updated from an older ownCloud or fresh install:**
Fresh Install

**Where did you install ownCloud from:**
Downloaded tarball from owncloud.org

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.0.0.0
  - dav: 0.1.5
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
  - user_ldap: 0.8.0
Disabled:
  - encryption
  - external
  - files_external
  - user_external
```

**The content of config/config.php:**

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost"
        ],
        "datadirectory": "\/usr\/local\/srv\/owncloud\/data\/owncloud",
        "overwrite.cli.url": "http:\/\/localhost",
        "dbtype": "pgsql",
        "version": "9.0.0.19",
        "dbname": "owncloud",
        "dbhost": "10.2.0.1",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "mail_domain": "waltonhoops.com",
        "mail_from_address": "owncloud",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "10.2.0.1",
        "mail_smtpport": "587",
        "mail_smtpsecure": "tls",
        "overwritehost": "cloud.waltonhoops.com",
        "overwriteprotocol": "https",
        "instanceid": "ocqjju513wie"
    }
}
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
LDAP
#### LDAP configuration (delete this part if not used)

```
+-------------------------------+---------------------------------------------+
| Configuration                 |                                             |
+-------------------------------+---------------------------------------------+
| hasMemberOfFilterSupport      | 0                                           |
| hasPagedResultSupport         |                                             |
| homeFolderNamingRule          |                                             |
| lastJpegPhotoLookup           | 0                                           |
| ldapAgentName                 |                                             |
| ldapAgentPassword             | ***                                         |
| ldapAttributesForGroupSearch  |                                             |
| ldapAttributesForUserSearch   |                                             |
| ldapBackupHost                |                                             |
| ldapBackupPort                |                                             |
| ldapBase                      | dc=waltonhoops,dc=com                       |
| ldapBaseGroups                | dc=waltonhoops,dc=com                       |
| ldapBaseUsers                 | dc=waltonhoops,dc=com                       |
| ldapCacheTTL                  | 600                                         |
| ldapConfigurationActive       | 1                                           |
| ldapDynamicGroupMemberURL     |                                             |
| ldapEmailAttribute            | mail                                        |
| ldapExperiencedAdmin          | 0                                           |
| ldapExpertUUIDGroupAttr       |                                             |
| ldapExpertUUIDUserAttr        |                                             |
| ldapExpertUsernameAttr        |                                             |
| ldapGroupDisplayName          | cn                                          |
| ldapGroupFilter               |                                             |
| ldapGroupFilterGroups         |                                             |
| ldapGroupFilterMode           | 0                                           |
| ldapGroupFilterObjectclass    |                                             |
| ldapGroupMemberAssocAttr      | uniqueMember                                |
| ldapHost                      | 10.2.0.1                                    |
| ldapIgnoreNamingRules         |                                             |
| ldapLoginFilter               | (&(|(objectclass=inetOrgPerson))(uid=%uid)) |
| ldapLoginFilterAttributes     |                                             |
| ldapLoginFilterEmail          | 0                                           |
| ldapLoginFilterMode           | 0                                           |
| ldapLoginFilterUsername       | 1                                           |
| ldapNestedGroups              | 0                                           |
| ldapOverrideMainServer        |                                             |
| ldapPagingSize                | 500                                         |
| ldapPort                      | 389                                         |
| ldapQuotaAttribute            |                                             |
| ldapQuotaDefault              |                                             |
| ldapTLS                       | 0                                           |
| ldapUserDisplayName           | cn                                          |
| ldapUserDisplayName2          |                                             |
| ldapUserFilter                | (|(objectclass=inetOrgPerson))              |
| ldapUserFilterGroups          |                                             |
| ldapUserFilterMode            | 0                                           |
| ldapUserFilterObjectclass     | inetOrgPerson                               |
| ldapUuidGroupAttribute        | auto                                        |
| ldapUuidUserAttribute         | auto                                        |
| turnOffCertCheck              | 0                                           |
| useMemberOfToDetectMembership | 1                                           |
+-------------------------------+---------------------------------------------+
```
### Client configuration

**Browser:**
Chrome

**Operating system:**
Windows 10, Arch Linux, Android 6

**Comment by RealRancor** [comment]
Something which could be catched with https://github.com/owncloud/core/issues/22941

**Comment by PVince81** [comment]
@DeepDiver1975 @georgehrke @blizzz

**Comment by blizzz** [comment]
should be fixed with 9.0.1, please update and retry

**Comment by PVince81** [comment]
Closing due to lack of feedback. Please reopen if the issue persists in 9.0.7 or 9.1.3.
